src/site_upload/api_gateway_authorizer/api_gateway_authorizer.py

```diff
@@ -4,23 +4,42 @@
 
 # pylint: disable=invalid-name,pointless-string-statement
 
+import json
 import os
 import re
 
-from src.handlers.shared.enums import BucketPath
-from src.handlers.shared.functions import get_s3_json_as_dict
+import boto3
+from botocore.exceptions import ClientError
 
 
 class AuthError(Exception):
     pass
 
 
+def get_secret():
+    """Retrieves a specified secret.
+
+    This is largely unmodified boilerplate from the secrets manager recommended approach
+    for fetching secrets, except for getting the values from environment variables"""
+
+    secret_name = os.environ.get("SECRET_NAME")
+    region_name = os.environ.get("REGION")
+
+    session = boto3.session.Session()
+    client = session.client(service_name="secretsmanager", region_name=region_name)
+
+    try:
+        get_secret_value_response = client.get_secret_value(SecretId=secret_name)
+    except ClientError as e:
+        raise e
```

Comment:
why this try/except block, if you're not doing anything with the error?

Reply:
this was the stock 'get the secrets' behavior, which i left in here mostly because this file is stock AWS code with a lil seasoning on top. but i could trim this.